### PR TITLE
Task 1782499: user input validation

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
for problem `Task 1783167: intellij calls validation method with non-stop once there is validation errors`, it will be solved in other PR

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureComboBox.java [hunk 1]
```diff
@@ -75,21 +85,21 @@ public void customize(final JList l, final Object o, final int i, final boolean
             }
         });
         if (isFilterable()) {
-            this.addPopupMenuListener(new AzureComboBoxPopupMenuListener());
+//            this.addPopupMenuListener(new AzureComboBoxPopupMenuListener());
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureComboBox.java:88]: why?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/ComboBoxResourceGroup.java [hunk 2]
```diff
@@ -45,8 +45,12 @@ protected String getItemText(final Object item) {
         return ((ResourceGroup) item).name();
     }
 
-    public void refreshWith(Subscription subscription) {
+    public void setSubscription(Subscription subscription) {
         this.subscription = subscription;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/ComboBoxResourceGroup.java:49]: two many duplicate setSubscription method

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureComboBox.java (hunk 1 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureComboBox.java]: by mistake

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/webapp/WebAppConfigFormPanelAdvanced.java [hunk 3]
```diff
@@ -91,25 +108,27 @@ public void setVisible(final boolean visible) {
...
-                this.selectorRegion.clear();
-            }
-        });
+        this.selectorServicePlan.addItemListener(this::onServicePlanChanged);
+        this.selectorSubscription.addItemListener(this::onSubscriptionChanged);
+    }
+
+    private void onSubscriptionChanged(final ItemEvent e) {
+        //TODO: @wangmi try subscription mechanism? e.g. this.selectorGroup.subscribe(this.selectSubscription)
+        if (e.getStateChange() == ItemEvent.SELECTED || e.getStateChange() == ItemEvent.DESELECTED) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/webapp/WebAppConfigFormPanelAdvanced.java:117]: If we change the subscription, will we get an DESELECTED event?

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/webapp/WebAppConfigFormPanelAdvanced.java:117]: no. 
this is to handle `ItemEvent.DESELECTED` event, I guess this event will be fired when user cancel select a item.
for this case, I need to tell the denpending component that subscription is cleared, you have to refresh your items and revalidate.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureTextField.java [hunk 4]
```diff
@@ -27,12 +27,19 @@
...
 
-public class AzureTextField extends JTextField implements AzureFormInputComponent<String> {
+public class AzureTextField extends JTextField implements AzureFormInputComponent<String>, DocumentListener {
     @Getter
     @Setter
     private boolean required;
 
+    public AzureTextField() {
+        super();
+        this.getDocument().addDocumentListener(this);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/AzureTextField.java:40]: create an new `DocumentAdapter` with `onValueChanged()`?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/ComboBoxResourceGroup.java (hunk 2 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/component/input/ComboBoxResourceGroup.java:49]: yes, but java doesn't support multiple class inheritance.
do you have any idea to solve this problem?

**Pull request merged**